src/edfi-paging-test/edfi_paging_test/api/request_client.py

```diff
@@ -0,0 +1,228 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from dataclasses import dataclass
+import os
+from typing import Any, Dict, List
+from http import HTTPStatus
+import socket
+
+from opnieuw import retry
+from requests import Response
+from requests.exceptions import ConnectionError, HTTPError, RequestException, Timeout
+from requests.auth import HTTPBasicAuth
+from urllib3.exceptions import ProtocolError
+from oauthlib.oauth2 import BackendApplicationClient  # type: ignore
+from oauthlib.oauth2 import TokenExpiredError  # type: ignore
+from requests_oauthlib import OAuth2Session  # type: ignore
+from edfi_paging_test.api.paginated_result import PaginatedResult
+
+
+REQUEST_RETRY_TIMEOUT_SECONDS = 60
+PERF_RESOURCE_LIST = list(
+    os.environ.get("PERF_RESOURCE_LIST") or ["StudentSectionAttendanceEvents"])
+OAUTH_TOKEN_URL = "/oauth/token/"
+
+
+@dataclass
+class RequestClient:
+    """
+    The RequestClient class wraps all the configuration complexity related
+    to authentication and http requests
+    Parameters
+    ----------
+    api_key : str
+        The API client key for EdFi.
+    api_secret : str
+        The API client secret for EdFi.
+    api_base_url : [str]
+        The API base url.
+    Attributes
+    ----------
+    oauth : OAuth2Session
+        The two-legged authenticated OAuth2 session.
+    """
+    api_key: str = str(os.environ.get("PERF_API_KEY"))
+    api_secret: str = str(os.environ.get("PERF_API_SECRET"))
+    api_base_url: str = str(os.environ.get("PERF_API_BASEURL"))
+    retry_count: int = int(os.environ.get("PERF_RETRY_COUNT") or 4)
+    page_size: int = int(os.environ.get("PERF_API_PAGE_SIZE") or 100)
```

Comment:
What if the user passes these in on the command line instead of through environment variables?

Reply:
We need these to be passed in as an argument to the constructor, rather than being read in from environment variables.